File: PhotonCore/src/main/java/com/outoftheboxrobotics/photoncore/HAL/Motors/Commands/PhotonLynxGetMotorChannelEnableCommand.java
package com.outoftheboxrobotics.photoncore.HAL.Motors.Commands;

import com.qualcomm.hardware.lynx.LynxModuleIntf;
import com.qualcomm.hardware.lynx.LynxNackException;
import com.qualcomm.hardware.lynx.commands.LynxDatagram;
import com.qualcomm.hardware.lynx.commands.LynxInterfaceResponse;
import com.qualcomm.hardware.lynx.commands.LynxMessage;
import com.qualcomm.hardware.lynx.commands.core.LynxGetMotorChannelEnableCommand;
import com.qualcomm.hardware.lynx.commands.core.LynxGetMotorChannelEnableResponse;
import com.qualcomm.hardware.lynx.commands.standard.LynxAck;
import com.qualcomm.hardware.lynx.commands.standard.LynxNack;
import com.qualcomm.robotcore.hardware.configuration.LynxConstants;

import java.nio.ByteBuffer;
import java.util.concurrent.CompletableFuture;

public class PhotonLynxGetMotorChannelEnableCommand<RESPONSE extends LynxMessage> extends LynxGetMotorChannelEnableCommand implements PhotonCommandBase{
    private CompletableFuture<LynxMessage> future;

    @Override
    public void onResponseReceived(LynxMessage response) {
        future.complete(response);
        super.onResponseReceived(response);
    }

    @Override
    public void onAckReceived(LynxAck ack) {
        future.complete(ack);
        super.onAckReceived(ack);
    }

    @Override
    public void onNackReceived(LynxNack nack) {
        future.complete(nack);
        super.onNackReceived(nack);
    }

    @Override
    public CompletableFuture<LynxMessage> getResponse() throws LynxNackException {
        return future;
    }

    @Override
    public void acquireNetworkLock() throws InterruptedException {
        //Nah, I'm gonna do my own thing
    }

    @Override
    public void releaseNetworkLock() throws InterruptedException {
        //Nah, I'm gonna do my own thing
    }

    public PhotonLynxGetMotorChannelEnableCommand(LynxModuleIntf module, int motorz) {
        super(module, motorz);
    }
}

File: PhotonCore/src/main/java/com/outoftheboxrobotics/photoncore/HAL/Motors/Commands/PhotonLynxGetMotorChannelEnableCommand.kt
package com.outoftheboxrobotics.photoncore.HAL.Motors.Commands

import com.qualcomm.hardware.lynx.LynxModuleIntf
import com.qualcomm.hardware.lynx.LynxNackException
import com.qualcomm.hardware.lynx.commands.LynxMessage
import com.qualcomm.hardware.lynx.commands.core.LynxGetMotorChannelEnableCommand
import com.qualcomm.hardware.lynx.commands.standard.LynxAck
import com.qualcomm.hardware.lynx.commands.standard.LynxNack
import java.util.concurrent.CompletableFuture


class PhotonLynxGetMotorChannelEnableCommand<RESPONSE : LynxMessage?>(module: LynxModuleIntf?, motorz: Int, override val response: CompletableFuture<LynxMessage?>?) : LynxGetMotorChannelEnableCommand(module, motorz), PhotonCommandBase {

    private val future: CompletableFuture<LynxMessage>? = null;

    override fun onResponseReceived(response: LynxMessage) {
        future?.complete(response)
        super.onResponseReceived(response)
    }

    override fun onAckReceived(ack: LynxAck) {
        response!!.complete(ack)
        super.onAckReceived(ack)
    }

    override fun onNackReceived(nack: LynxNack) {
        response!!.complete(nack)
        super.onNackReceived(nack)
    }
}